### PR TITLE
Replace makefile with just

### PR DESCRIPTION
**Pull request by pitasi**
Install `just` by running `brew install just` (on Mac OS) or see https://just.systems/man/en/chapter_4.html for more instructions.

`just` has a few advantages over `make`, it's simpler to write and read and has less unexpected behaviours (since `just` is a command runner and not a build system).

I migrated the existing makefile targets to just, and added a few more ones.

## Summary by CodeRabbit

- **New Features**
	- Introduced the use of `just` for building and installing software, replacing `make`.
	- Added new build and release processes using Just, Docker, and Goreleaser.
	- Updated installation and local chain setup instructions across various documentation.
	- Enhanced project management with new `justfile` configurations in multiple modules.

- **Documentation**
	- Updated terminology in `.spelling` and various documents to reflect new tools and processes.
	- Modified the whitepaper to standardize spelling from "neutralising" to "neutralizing."

- **Refactor**
	- Streamlined the release process and simplified project management scripts.

### CODE REVIEW

**Review by jjheywood — APPROVED** [review]

File: cmd/justfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+app_name := "warden"
+binary_name := app_name + "d"
+chain_id := app_name
+tag := `git describe --tags --always`
+commit := `git rev-parse HEAD`
+dirty := `git diff --quiet || echo "-dirty"`
+version := tag + dirty
+ldflags := "-ldflags \"-s -w -X github.com/cosmos/cosmos-sdk/version.Name=" + app_name + " -X github.com/cosmos/cosmos-sdk/version.AppName=" + app_name + "d -X github.com/cosmos/cosmos-sdk/version.Version=" + version + " -X github.com/cosmos/cosmos-sdk/version.Commit=" + commit + " -X github.com/warden-protocol/wardenprotocol/cmd/wardend/cmd.ChainID=" + chain_id + "\""
```
Comment (jlehtimaki) [inline cmd/justfile:8]: Could you use '-ldflags .......' to surround this to make it more readable?

File: docs/docs/developers/installation.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ installing each one of them.
 ## Prerequisites
 
 - [Go](https://golang.org/dl/) 1.22 or later
-- `make`
+- `just` (https://just.systems/man/en/chapter_4.html)
```
Comment (jlehtimaki) [inline docs/docs/developers/installation.md:14]:
```suggestion
- [just](https://just.systems/man/en/chapter_4.html)
```

File: justfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    	-e CGO_ENABLED=1 \
+    	-v /var/run/docker.sock:/var/run/docker.sock \
+    	-v {{ invocation_directory() }}:/go/src/wardend \
+    	-w /go/src/wardend \
+    	ghcr.io/goreleaser/goreleaser:v1.25.1 \
+    	--clean --skip=publish -f ./.goreleaser-amd64.yaml
+
+release_tag := `git tag --points-at HEAD`
+
+# use goreleaser to build binaries, then create a GitHub release.
```
Comment (jlehtimaki) [inline justfile:34]: Why not do release with `goreleaser`?

Comment (pitasi) [reply justfile:34]: I'm using goreleaser and goreleaser-cross, then I'm "merging" the two outputs into a single GitHub release, which is something that can only be done with the paid version of goreleaser

**Pull request merged**